Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -143,21 +207,22 @@ namespace AzToolsFramework
         for (auto propertyWidgetIter = m_widgetToPropertyHandler.begin(), endIter = m_widgetToPropertyHandler.end();
              propertyWidgetIter != endIter; ++propertyWidgetIter)
         {
-            m_columnLayout->removeWidget(propertyWidgetIter->first);
+            QWidget* propertyWidget = propertyWidgetIter->first;
+            auto toRemove = AZStd::remove(m_domOrderedChildren.begin(), m_domOrderedChildren.end(), propertyWidget);
+            m_domOrderedChildren.erase(toRemove, m_domOrderedChildren.end());
+            m_columnLayout->removeWidget(propertyWidget);
         }
         m_widgetToPropertyHandler.clear();
 
-        DestroyLayoutContents(m_columnLayout);
-
-        // delete all child rows, this will also remove them from the layout
+        // delete all remaining child widgets, this will also remove them from their layout
         for (auto entry : m_domOrderedChildren)
         {
-            DPERowWidget* rowChild = qobject_cast<DPERowWidget*>(entry.data());
-            if (rowChild)
+            if (entry)
             {
-                rowChild->deleteLater();
+                delete entry;
```

Comment:
Any reason why we're not doing `deleteLater` here? I don't know that it's a specific problem for row widgets, but it might be safer to leave this as-is.

Reply:
just because it was unnecessary and served no real purpose. I think you once asked me what I was hiding by doing a deleteLater and so I decided to remove them :P